src/adaptor.py

```diff
@@ -215,6 +215,11 @@ def handler(json_request, outgoing):
             if conf.SEND_LAX_PATCHED_AJSON: # makes in-place changes to the data
                 validate.add_placeholders_for_validation(article_data)
                 LOG.info("placeholders attached")
+
```

Comment:
I think this should be pushed into `render_single` as the lowest common denominator between backfills and adaptor ingests; that would make explicit how to always provide a location, see my long comment.